src/main/java/com/github/kongchen/swagger/docgen/reader/JaxrsReader.java

```diff
@@ -99,6 +100,7 @@ protected Swagger read(Class<?> cls, String parentPath, String parentMethod, boo
             swagger = new Swagger();
         }
         Api api = AnnotationUtils.findAnnotation(cls, Api.class);
+        ApplicationPath apiApplicationPath = AnnotationUtils.findAnnotation(cls, ApplicationPath.class);
```

Comment:
If I understand this line correctly, we search for the ApplicationPath annotation on the Resource class _cls_?
I doubt this will work in every possible case, and even matches spec. [1]
According to the ApplicationPath annotation itself ```May only be applied to a subclass of Application.```. It is therefore not allowed on Resource classes.
You can also define multiple Application Subclasses according to the spec [2].

A complete (although untested) example of what is allowed:
```
@ApplicationPath("pets/")
public class PetApplication extends Application {
    @Override
    public Set<Class<?>>getClasses() {
        return Sets.newHashSet(PetsResource.class);
    }
}

public class PetsResource {
    @GET
    public void listPets(String someFilter) {
        ....
    }
}

@ApplicationPath("")
public class CarApplication extends Application {
    @Override
    public Set<Class<?>>getClasses() {
        return Sets.newHashSet(CarsResource.class);
    }
}

@Path("/cars/")
public class CarsResource{
    @Path("{id}")
    @GET
    public void get(Long id) {
        ....
    }
}
```

[1] https://docs.oracle.com/javaee/7/api/javax/ws/rs/ApplicationPath.html#value--
[2] https://issues.jboss.org/browse/RESTEASY-650
[3] SR-000370 Java API for RESTful Web Services 2.1 Specification Final Release for Evaluation